storage/data-plane/src/test/java/com/microsoft/azure/storage/BlockBlobAPITest.groovy

```diff
@@ -165,19 +175,25 @@ class BlockBlobAPITest extends APISpec {
         bu2.stageBlockFromURL(blockID, bu.toURL(), null).blockingGet().statusCode() == 201
     }
 
-    @Unroll
-    def "Stage block from URL IA"() {
+    def "Stage block from URL IA without BlockID"() {
+        setup:
+        def blockId = getBlockID()
+
         when:
-        bu.stageBlockFromURL(blockID, sourceURL, null, null, null, null, null)
+        bu.stageBlockFromURL(null, new URL("http://www.example.com"), null, null, null, null, null)
```

Comment:
This one is because getBlockId in where clause did not invoke the @Rule (testName) in APISpec. So it failed by complaining of invoking the testName.getMethodName() on null object.

So I have to put getBlockId in when clause instead of where.